### PR TITLE
Reports: The reporting contact id is added

### PR DESCRIPTION
**Pull request by annando**
Reports from remote systems obviously can't contain a user id. Because of that we now add the contact id of the reporter as well.

### CODE REVIEW

**Review by MrPetovan — COMMENTED** [review]
Since local users also have a contact id, shouldn't we do away with the `uid` completely?

**Comment by annando** [comment]
I thought about this as well. But on the other hand this will help with possible extensions like that a user could have a list of their reports. Also this is an easy distinction between local and remote reports.

File: src/Moderation/Entity/Report.php [hunk 1]
```diff
@@ -35,24 +35,30 @@ class Report extends \Friendica\BaseEntity
...
+	protected $category;
 	/** @var bool Whether this report should be forwarded to the remote server */
 	protected $forward;
 	/** @var \DateTime|null When the report was created */
 	protected $created;
 	/** @var array Optional list of URI IDs of posts supporting the report*/
 	protected $postUriIds;
 
-	public function __construct(int $uid, int $cid, \DateTime $created, string $comment = '', bool $forward = false, array $postUriIds = [], int $id = null)
+	public function __construct(int $uid = null, int $reporterId, int $cid, \DateTime $created, string $comment = '', string $category = null, bool $forward = false, array $postUriIds = [], int $id = null)
```
Comment (MrPetovan) [inline src/Moderation/Entity/Report.php]: If you intent on keeping the `uid` field around, please move the optional parameter after the mandatory ones.
```suggestion
	public function __construct(int $reporterId, int $cid, \DateTime $created, int $uid = null, string $comment = '', string $category = null, bool $forward = false, array $postUriIds = [], int $id = null)
```
Please be sure to update the places the constructor is called.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Moderation/Entity/Report.php (hunk 1 above)
Comment (annando) [reply src/Moderation/Entity/Report.php]: done

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**